sunkit_instruments/suvi/suvi.py

```diff
@@ -0,0 +1,1122 @@
+        not found in the filename, it will not be recognized.
+
+    .. note::
+       GOES-16 L2 Thematic Maps are available
+       `here. <https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/data/suvi-l2-thmap/>`__
+
+    Parameters
+    ----------
+    input_filename: `str`
+        File to read.
+
+    timestamp: `bool`, optional. Default: True. 
+        If True, plots the timestamp of the observation. 
+
+    legend: `bool`, optional. Default: True. 
+        If True, plots a legend with the different classes.
+
+    figsize: `tuple`, optional. Default: (10, 10).
+        The size of the figure. Should not be smaller than 
+        (7, 7) to look decent.
+    """
+
+    if '-l2-thmap' in os.path.basename(input_filename):
+        hdu = fits.open(input_filename)
+        thmap_data = hdu[0].data
+        if timestamp:
+            time_st = hdu[0].header['DATE-OBS'][0:19]
+        hdu.close()
+        
+        SOLAR_CLASSES = [('unlabeled', 0),
+                        ('outer_space', 1),
+                        ('bright_region', 3),
+                        ('filament', 4),
+                        ('prominence', 5),
+                        ('coronal_hole', 6),
+                        ('quiet_sun', 7),
+                        ('limb', 8),
+                        ('flare', 9)]
+            
+        SOLAR_CLASS_NAME = {number: theme for theme, number in SOLAR_CLASSES}
+        
+        SOLAR_COLORS = {"unlabeled": "white",
+                        "outer_space": "black",
+                        "bright_region": "#F0E442",
+                        "filament": "#D55E00",
+                        "prominence": "#E69F00",
+                        "coronal_hole": "#009E73",
+                        "quiet_sun": "#0072B2",
+                        "limb": "#56B4E9",
+                        "flare": "#CC79A7"}
+            
+        colortable = [SOLAR_COLORS[SOLAR_CLASS_NAME[i]] if i in SOLAR_CLASS_NAME else 'black'
+                      for i in range(max(list(SOLAR_CLASS_NAME.keys())) + 1)]
+        cmap = ListedColormap(colortable)
+
+        fig, ax = plt.subplots(constrained_layout=True, figsize=figsize)
+        ax.imshow(thmap_data, origin='lower', cmap=cmap, vmin=-1, vmax=len(colortable), interpolation='none')
+        ax.set_axis_off()
+        if timestamp:
+            ax.text(10, 1245, time_st, fontsize=14, color='white')
+        if legend:
+            legend_elements = [Patch(facecolor=color, edgecolor="black", label=label.replace("_", " "))
+                               for label, color in SOLAR_COLORS.items()]
+            ax.legend(handles=legend_elements, loc='upper right', ncol=3, fancybox=True, shadow=False)
+        fig.show()
+    else:
+        raise ValueError("File "+input_filename+" does not look like a SUVI L2 Thematic Map.")
+
+
+def read_suvi(input_filename, return_DQF=False, return_header_only=False):
```

Comment:
I do wonder if we should spin out the read functions to a `io.py` file just to keep the logic separate?

Reply:
That's up to you, and the philosophy behind the code. Personally, I prefer not having to type several import commands, but I do get the point about keeping the logic separate. Let me know what you would prefer.

Reply:
I would prefer if the files were shorter, it makes it easier to review and maintain.

We can hide the fact its in an io file by importing it in `__init__.py``.